feat: add cloudfront-s3-cdk-go pattern

Gryczka opened the pull request:
*Issue #, if available:* #2250

*Description of changes:* Adding a new pattern for CloudFront to S3 Static Site implemented using the CDK in Go.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

pputhran commented on cloudfront-s3-cdk-go/example-pattern.json:
```diff
@@ -0,0 +1,57 @@
+        "name": "Patrick Gryczka",
+        "image": "https://www.gryczka.net/img/patryk.jpg",
+        "bio": "Patrick Gryczka is a Senior Solutions Architect at AWS specializing in Serverless technologies and the Sports industry.",
+        "linkedin": "https://www.linkedin.com/in/gryczka/"
```
Please remove the URL in this property, only include the ID to your LinkedIn profile.

pputhran commented on cloudfront-s3-cdk-go/example-pattern.json:
```diff
@@ -0,0 +1,57 @@
+    "gitHub": {
+      "template": {
+        "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cloudformation-s3-cdk-go",
+        "templateURL": "serverless-patterns/cloudformation-s3-cdk-go",
```
Please remove the projectFolder value from the templateFile. Please refer to some of the other Go examples that we have under Serverless patterns. 
```
        "template": {
            "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cloudformation-s3-cdk-go",
            "templateURL": "serverless-patterns/cloudformation-s3-cdk-go",
            "projectFolder": "cloudfront-s3-cdk-go",
            "templateFile": "cloudformation-s3-cdk-go.go"
        }
        
        ```

Gryczka replied:
Hi @pputhran I have removed the projectFolder value and fixed a few cloudformation/cloudfront typos.  

But could you point me toward one of the relevant go examples?  None of the go cdk examples I see have an example-pattern.json present.

pputhran reviewed (changes requested):
Please refer to the requested changes

Gryczka commented:
Hi @pputhran I have removed the projectFolder value and fixed a few cloudformation/cloudfront typos.

But could you point me toward one of the relevant go examples? None of the go cdk examples I see have an example-pattern.json present.

Gryczka commented:
Hi @pputhran just pinging back on whether you have a cdk go example json you could share.

pputhran commented on cloudfront-s3-cdk-go/example-pattern.json:
```diff
@@ -0,0 +1,56 @@
+      ]
+    },
+    "gitHub": {
+      "template": {
```
The projectFolder is the required field.  Also since the cloudfront-s3-cdk-go.go was renamed, its throwing a new error. Can you rename this file to cloudfronts3-cdk-go.go?

Can you please update as below

```
        "template": {
            "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/cloudformation-s3-cdk-go",
            "templateURL": "serverless-patterns/cloudformation-s3-cdk-go",
            "projectFolder": "cloudfront-s3-cdk-go",
            "templateFile": "cloudfronts3-cdk-go.go"
        }
```

pputhran commented:
> Hi @pputhran I have removed the projectFolder value and fixed a few cloudformation/cloudfront typos.
> 
> But could you point me toward one of the relevant go examples? None of the go cdk examples I see have an example-pattern.json present.

You are right. I assumed there were a few examples already. My bad. Apologies. I have added my comments and if you can update them, I can then approve it.

pputhran commented:
Any updates on this please?
